Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs

```diff
@@ -14,19 +17,30 @@ public SubmissionsRepository(IOjsDbContext context)
         {
         }
 
-        public IQueryable<Submission> AllPublic()
-        {
-            return this.All()
-                .Where(x => x.IsPublic ??
-                    (((x.Participant.IsOfficial && x.Problem.Contest.ContestPassword == null) ||
-                     (!x.Participant.IsOfficial && x.Problem.Contest.PracticePassword == null))
-                    && x.Problem.Contest.IsVisible && !x.Problem.Contest.IsDeleted
-                    && x.Problem.ShowResults));
-        }
+        public IQueryable<Submission> AllPublic() => 
+            this.All()
+                .Where(x => x.IsPublic 
+                    ?? (((x.Participant.IsOfficial && x.Problem.Contest.ContestPassword == null) ||
+                    (!x.Participant.IsOfficial && x.Problem.Contest.PracticePassword == null)) &&
+                     x.Problem.Contest.IsVisible && !x.Problem.Contest.IsDeleted &&
+                     x.Problem.ShowResults));
 
-        public Submission GetSubmissionForProcessing()
+        public IQueryable<Submission> AllForLecturer(string lecturerId)
         {
-            var submission =
+            var problemsIds = new HashSet<int>(
```

Comment:
Why use a HashSet instead of List?

Reply:
I use HashSet for better speed performance, because I make this collection only to check if an Id is contained in it. The Contains method for the HashSet is with complexity O(1) against O(n) for the List.